perfaware/sim86/shared/Sim8086.cpp

```diff
@@ -360,5 +391,5 @@ int main(int ArgCount, char** Args)
             std::cout << OutputBuffer;
             PrintFlags(FlagArray);
```

Comment:
I note that you do not reset registers and memory at the end of the loop. In my opinion, you should do it. Right now you might have different results for a given file depending on where it is in the list of arguments. For example, imagine if your program starts with `mov ax, [some memory]`, well that memory might well have different contents if you run it after another program.

Reply:
Registers are currently reset between each file (initialized at line 345). 
What would be better for memory though?
1. Explicitly 0 everything in a loop at the beginning
2. Increase stack size and initialize it on each file like Registers
3. Something else?

Reply:
Nice, I missed it for the registers.

For the memory:
1. Yes that's good.
2. It's not stored on the stack so it has no impact on this. The stack is used for local variables (basically the stack is how functions are implemented), not for globals (functions work independently of global variables, these are stored elsewhere in memory).

Reply:
It would be on the stack if I declared and initialized it at the same place as the register so that's why I asked.
Array zero'd here, though I'm worried about efficiency. I've read about memset() or std::fill, any thoughts?
3ed59084c529a00ceca6104eb774e7a5288f4171

Reply:
Oh right, then yeah it would have been on the stack, but that's not very good for 1 MB data structures.

`memset` is what is used typically. Also I would do this next to where you zero the registers, it goes together conceptually.

Reply:
Thanks
ec73a63efc12e0753fe6a3a38adf481b122b518e